src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookListener.java

```diff
@@ -13,6 +14,9 @@
 
     public static final Logger LOGGER = Logger.getLogger(GitLabWebHookListener.class.getName());
 
+    // Delay used for triggers to avoid GitLab cache
+    private static final long TRIGGER_DELAY_SECONDS = 30;
```

Comment:
It might be a good idea to make this parameter configurable..

Reply:
I was thinking about it, but then decided not to have it configurable because also for GitLab the cache timeout is hard-coded to be 30 seconds. As no other value than the GitLab hard-coded timeout would make sense here, I was thinking there would possibly be less confusion if users could not configure it.

Reply:
Oh, I see. Then let's wait for merging :)

Reply:
Okay, let's wait then. Thanks for reviewing!

Reply:
Considering the [discussion from Gitlab team](https://gitlab.com/gitlab-org/gitlab/-/issues/333548), it seems like a temporary solution.

Secondly, the delay issue only affects parts of the Gitlab versions. Let assume there is one user who has an old Gitlab, then his jobs cannot be triggered in time.

So, I prefer to have two changes here:

* Make the timeout value to be configurable in case of Gitlab changes it
  * Make it be an advanced configuration item, might be easier for some users. Normally, people don't need to change it.
* Don't have a timeout value if it's an old Gitlab. You can get the version from Gitlab API, then cache it.

Reply:
- Made the timeout configurable
- I tried to implement GitLab API version check to disable the trigger delay for old versions but failed because I could not find a suitable place to retrieve versions from GitLab API and cache them. Retrieving API version from WebHookListener gives an exception because WebHookListener is not allowed to access GitLab credentials.